usecases/schema/add_property.go

```diff
@@ -84,9 +87,31 @@ func (m *Manager) addClassProperty(ctx context.Context,
 	return m.addClassPropertyApplyChanges(ctx, className, prop)
 }
 
-func (m *Manager) setNewPropDefaults(class *models.Class, prop *models.Property) {
+func (m *Manager) setNewPropDefaults(class *models.Class, prop *models.Property) error {
 	m.setPropertyDefaults(prop)
+	if err := validateUserProp(class, prop); err != nil {
+		return err
+	}
 	m.moduleConfig.SetSinglePropertyDefaults(class, prop)
+	return nil
+}
+
+func validateUserProp(class *models.Class, prop *models.Property) error {
+	if prop.ModuleConfig != nil {
+		modconfig, ok := prop.ModuleConfig.(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("%v property config invalid", prop.Name)
+		}
+		vectorizerConfig, ok := modconfig[class.Vectorizer]
+		if !ok {
+			return fmt.Errorf("%v vectorizer module not part of the property", class.Vectorizer)
+		}
+		_, ok = vectorizerConfig.(map[string]interface{})
+		if !ok {
+			return errors.New("invalid module config")
```

Comment:
I think we can improve this, eg `vectorizerConfig %v not of type map[string]interface{}` or something

Reply:
Fixed.